File: Programmers/Level 3/4.cpp
#include <string>
#include <vector>

using namespace std;
bool visit[210];

void dfs(int n, int node, vector<vector<int>> computers) {
    visit[node] = true;
    for(int i=0; i<n; i++) {
        if (computers[node][i] == 1 && !visit[i])
            dfs(n, i, computers);
    }
}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    
    for(int i=0; i<n; i++) {
        if (!visit[i]){
            dfs(n, i, computers);
            answer++;
        }
    }
    
    return answer;
}

File: Programmers/Level 3/4.py
visited = [False] * 201

def dfs(node, n, computers):
    visited[node] = True
    
    for i in range(n):
        if computers[node][i] == 1 and visited[i] == False:
            dfs(i, n, computers)

def solution(n, computers):
    answer = 0
    for i in range(n):
        if visited[i]: continue
        dfs(i, n, computers)
        answer += 1
    return answer
